src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -330,16 +346,50 @@ private function getMutatorMethod($class, $property)
     /**
      * Extracts a property name from a method name.
      *
-     * @param string $methodName
+     * @param string                $methodName
+     * @param \ReflectionProperty[] $reflectionProperties
      *
      * @return string
      */
-    private function getPropertyName($methodName)
+    private function getPropertyName($methodName, array $reflectionProperties)
     {
         $pattern = implode('|', array_merge(self::$accessorPrefixes, self::$mutatorPrefixes));
 
         if (preg_match('/^('.$pattern.')(.+)$/i', $methodName, $matches)) {
+            if (!in_array($matches[1], self::$arrayMutatorPrefixes)) {
+                return $matches[2];
+            }
+
+            foreach ($reflectionProperties as $reflectionProperty) {
+                foreach ($this->getSingulars($reflectionProperty->name) as $name) {
+                    if ($name === lcfirst($matches[2])) {
+                        return $reflectionProperty->name;
+                    }
+                }
+            }
+
             return $matches[2];
         }
     }
+
+    /**
+     * Gets singulars of a word.
+     *
+     * @param string $plural
+     *
+     * @return array|null Returns null if cannot guess.
+     */
+    private function getSingulars($plural)
+    {
+        if (!class_exists('Symfony\Component\PropertyAccess\StringUtil')) {
+            return;
+        }
+
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class '\Symfony\Component\PropertyAccess\StringUtil' in method 'getSingulars'.](https://www.codacy.com/app/git_9/symfony/file/2331969670/issues/source?bid=3207821&fileBranchId=3207838#l388)